### PR TITLE
[DPEDE-2373] Add Radio Button web component in documentation examples

### PR DESCRIPTION
**Pull request by xft-dev**
https://ctl.atlassian.net/browse/DPEDE-2373

### CODE REVIEW

**Review by wojciechwlodarczyklumen — CHANGES_REQUESTED** [review]
image

I think (thats how it is on prod), the letters should be black. -> https://assets.ctl.io/chi/6.1.0/components/forms/radio-button/?theme=Lumen#error

Besides that looks ok :)

**Comment by xft-dev** [comment]
> image
> 
> I think (thats how it is on prod), the letters should be black. -> https://assets.ctl.io/chi/6.1.0/components/forms/radio-button/?theme=Lumen#error
> 
> Besides that looks ok :)

Thanks. I removed stateClass from Label in RadioButton.tsx (custom elements). So, text should be black.

**Review by josecarlosgonzalezv — APPROVED** [review]

**Review by wojciechwlodarczyklumen — APPROVED** [review]

File: src/website/views/components/forms/radio-button/_examples.pug [hunk 1]
```diff
@@ -236,22 +273,40 @@ p.-text
...
+            </section>
+          </div>
+        </div>
+        <div class="chi-form__item">
+          <chi-radio-button id="example_help_opt_1" label="Option 1" name="radios"></chi-radio-button>
+        </div>
+        <div class="chi-form__item">
+          <chi-radio-button id="example_help_opt_2" label="Option 2" name="radios"></chi-radio-button>
+        </div>
+      </fieldset>
```
Comment (emerson-pereira) [inline src/website/views/components/forms/radio-button/_examples.pug:308]: The help icon is not working here. Can we use chi-label `info-icon` / `info-icon-message`?

```js
<chi-label ... info-icon info-icon-message="Helpful information goes here.">Label</chi-label>
```

File: src/website/views/components/forms/radio-button/_examples.pug [hunk 2]
```diff
@@ -5,35 +5,41 @@ h3 Base
...
-      .chi-form__item.-mb--1
-        .chi-radio
-          input(class="chi-radio__input", type="radio", name="radios", id="radio-ba1")
-          label.chi-radio__label(for="radio-ba1") Option 1
-      .chi-form__item
-        .chi-radio
-          input(class="chi-radio__input", type="radio", name="radios", id="radio-ba2")
-          label.chi-radio__label(for="radio-ba2") Option 2
+      div.chi-form__item
+        chi-radio-button(id="example_base_opt_1" label="Option 1" name="radios")
```
Comment (emerson-pereira) [inline src/website/views/components/forms/radio-button/_examples.pug:9]: Id names in /website is missing one underline e.g. /documentation:

`example__base_opt_1`

File: src/documentation/pages/components-wip/forms/radio-button/examples/_base.vue [hunk 3]
```diff
@@ -26,30 +22,38 @@ import { Vue } from 'vue-facing-decorator';
...
+          disabled: false,
           id: 'htmlblueprint',
           label: 'HTML Blueprint',
         },
       ],
       codeSnippets: {
-        webcomponent: ``,
+        webcomponent: `<fieldset>
+  <legend class="chi-label">Select an option</legend>
+  <div class="chi-form__item>
```
Comment (emerson-pereira) [inline src/documentation/pages/components-wip/forms/radio-button/examples/_base.vue:38]: There many occurrences of this:

```suggestion
  <div class="chi-form__item">
```

File: src/documentation/pages/components-wip/forms/radio-button/examples/_optional.vue [hunk 4]
```diff
@@ -30,33 +26,45 @@ import { Vue } from 'vue-facing-decorator';
...
         },
       ],
       codeSnippets: {
-        webcomponent: ``,
+        webcomponent: `<fieldset>
+  <legend class="chi-label">
+    Select an option
+    <abbr class="chi-label__optional" aria-label="Optional field">(optional)</abbr>
+  </legend>
+  <legend class="chi-label">Select an option</legend>
```
Comment (emerson-pereira) [inline src/documentation/pages/components-wip/forms/radio-button/examples/_optional.vue:45]:
```suggestion
```

File: src/documentation/pages/components-wip/forms/radio-button/examples/_required.vue [hunk 5]
```diff
@@ -30,33 +26,45 @@ import { Vue } from 'vue-facing-decorator';
...
         },
       ],
       codeSnippets: {
-        webcomponent: ``,
+        webcomponent: `<fieldset>
+  <legend class="chi-label">
+    Select an option
+    <abbr class="chi-label__required" aria-label="Required field">*</abbr>
+  </legend>
+  <legend class="chi-label">Select an option</legend>
```
Comment (emerson-pereira) [inline src/documentation/pages/components-wip/forms/radio-button/examples/_required.vue:45]:
```suggestion
```

File: src/documentation/pages/components-wip/forms/radio-button/examples/_help.vue [hunk 6]
```diff
@@ -39,18 +35,38 @@ declare const chi: any;
...
+      disabled: false,
       id: 'htmlblueprint',
       label: 'HTML Blueprint',
     },
   ];
   codeSnippets = {
-    webcomponent: ``,
+    webcomponent: `<fieldset>
+  <div class="chi-label__wrapper">
+    <legend class="chi-label">Select an option</legend>
```
Comment (emerson-pereira) [inline src/documentation/pages/components-wip/forms/radio-button/examples/_help.vue:51]: Also here?

```js
<chi-label ... info-icon info-icon-message="Helpful information goes here.">Label</chi-label>
```

File: src/documentation/pages/components-wip/forms/radio-button/examples/_error.vue [hunk 7]
```diff
@@ -32,33 +28,44 @@ import { Vue } from 'vue-facing-decorator';
...
+    Select an option
+    <abbr class="chi-label__required" aria-label="Required field">*</abbr>
+  </legend>
+  <div class="chi-form__item>
+    <chi-radio-button id="example__error_opt_1" label="Option 1" name="radios" state="danger" required></chi-radio-button>
+  </div>
+  <div class="chi-form__item>
+    <chi-radio-button id="example__error_opt_2" label="Option 2" name="radios" state="danger" required></chi-radio-button>
+  </div>
+</fieldset>`,
```
Comment (emerson-pereira) [inline src/documentation/pages/components-wip/forms/radio-button/examples/_error.vue:53]: Missing snippet here?

```js
<div class="chi-label -status -danger">
  <i class="chi-icon icon-circle-warning" aria-hidden="true"></i>
  Please select an option
</div>
```

File: src/documentation/pages/components-wip/forms/radio-button/examples/_required.vue (hunk 5 above)
Comment (emerson-pereira) [inline src/documentation/pages/components-wip/forms/radio-button/examples/_required.vue:45]:
```suggestion
```

**Review by emerson-pereira — CHANGES_REQUESTED** [review]

File: src/documentation/pages/components-wip/forms/radio-button/examples/_required.vue (hunk 5 above)
Comment (xft-dev) [reply src/documentation/pages/components-wip/forms/radio-button/examples/_required.vue:45]: Duplicated

File: src/documentation/pages/components-wip/forms/radio-button/examples/_error.vue (hunk 7 above)
Comment (xft-dev) [reply src/documentation/pages/components-wip/forms/radio-button/examples/_error.vue:53]: It looks ok. I don't understand what you mean.

File: src/documentation/pages/components-wip/forms/radio-button/examples/_help.vue (hunk 6 above)
Comment (xft-dev) [reply src/documentation/pages/components-wip/forms/radio-button/examples/_help.vue:51]: chi-label doesn't work for me

File: src/website/views/components/forms/radio-button/_examples.pug (hunk 1 above)
Comment (xft-dev) [reply src/website/views/components/forms/radio-button/_examples.pug:288]: I've changed but it works for me

**Review by emerson-pereira — APPROVED** [review]

**Pull request merged**